dfttk/debye.py

```diff
@@ -0,0 +1,348 @@
+    .. math::
+       D(x) = \\frac{\\pi^4}{5x^3} - 3 \\sum{k=1} \\frac{1}{k} (1 + \\frac{3}{kx} + \\frac{6}{k^2x^2} + \\frac{6}{k^3x^3}) * e^{-kx}
+        
+    See references,
+    Gonzalez, I., Kondrashuk, I., Moll, V. H., & Vega, A. Analytic Expressions for Debye Functions and the Heat Capacity of a Solid. Mathematics, 10(10), 1745. https://doi.org/10.3390/math10101745
+    Abramowitz, M. and Stegun, I.A. eds., 1968. Handbook of mathematical functions with formulas, graphs, and mathematical tables (Vol. 55). US Government printing office.
+    Khishchenko, K., Analytic approximation of the Debye function, Mathematica Montisnigri (vol. 49), 2020.  https://doi.org/10.20948/MATHMONTIS-2020-49-8
+
+    Args:
+        x_array: array of input values for the debye function
+        prec: Precission. Terminates the series expansion when the absolute value of the term is less than prec
+        nth_bernoulli: Determines the nth Bernoulli number to calculate. A list of bernoulli numbers is generated prior calculating the series expansion. There should be no reason to change this value under normal circumstances.
+        
+    Raises:
+        ValueError: If the precision is not between 0 and 1
+        ValueError: If x < -2𝜋
+        IndexError: If the bernoulli number at index 2k is not available. This indicates slow converges of the Debye function series expansion. If you wish to calculate values for x < -𝜋 convergence may be slow and you may need to increase nth_bernoulli.
+
+    Returns:
+        np.array: The value of the debye function evaluated at each x in x_array
+    """
+    if not 0 < prec < 1:
+        raise ValueError("The precision must be between 0 and 1")
+    result = np.zeros_like(x_array)
+    bern_list = bernoulli(nth_bernoulli) # 2*k must be less than 100
+    for i, x in enumerate(x_array):
+        term = 1 # ensures the while loop runs at least once
+        k = 1
+        if x >= 0.7*np.pi:
+            summation = np.pi**4/(5*x**3)
+            while abs(term) > prec:
+                term = -3*(1/k*(1 + 3/(k*x) + 6/(k**2*x**2) + 6/(k**3*x**3))*np.exp(-k*x))
+                summation += term
+                k += 1
+            result[i] = summation
+        elif -2*np.pi < x < 0.7*np.pi:
+            summation = 1 - 3/8*x
+            while abs(term) > prec:
+                try:
+                    term = 3 * (bern_list[2*k]/((2*k+3)*gamma(2*k+1)) * x**(2*k))
+                except IndexError:
+                    raise IndexError(f"IndexError: the bernoulli number at index {2*k} is not available. This indicates slow converges of the Debye function series expansion. If you wish to calculate values for x < -𝜋 convergence may be slow and you may need to increase nth_bernoulli.")
+                summation += term
+                k += 1
+            result[i] = summation
+        else:
+            raise ValueError("The debye function series expansions used are only valid for x > -2𝜋")
+    return result
+
```

Comment:
Maybe we should switch the variable names term and summation. Because the summation is the constant and the terms are the ones being summed.

Reply:
Why was n=3 chosen again?

Reply:
> Maybe we should switch the variable names term and summation. Because the summation is the constant and the terms are the ones being summed.

No. Summation is not a constant. It is the sum of each term. See lines 112 and 122.

I see your confusion though as step in the summation has multiple terms.
![image](https://github.com/user-attachments/assets/e6bc16b2-5349-4979-bd96-16a9e6a8dcff)


> Why was n=3 chosen again?

We use the third-order Debye function because there are three polarizations in real space, two transverse and one longitudinal.